memeolist/app/src/main/java/org/aerogear/android/app/memeolist/ui/MemeolistApplication.java

```diff
@@ -23,8 +23,10 @@ public void onCreate() {
         authService = new AuthService(authServiceConfig);
 
         if(authService.currentUser() != null) {
+            // FIXME This is serious bug (profile should change at any time and should not be cached)
```

Comment:
Can you be more clear about the bug? I can't see a problem here. Keep in mind the [UserProfile](https://github.com/aerogear/aerogear-android-example-apps/blob/master/memeolist/app/src/main/java/org/aerogear/android/app/memeolist/model/UserProfile.java) is not a [AeroGear Auth UserPrincipal](https://github.com/aerogear/aerogear-android-sdk/blob/master/auth/src/main/java/org/aerogear/mobile/auth/user/UserPrincipal.java). Anyway every login this property is [updated](https://github.com/aerogear/aerogear-android-example-apps/blob/master/memeolist/app/src/main/java/org/aerogear/android/app/memeolist/ui/LoginActivity.java#L54).

Reply:
I will remove this and check the logout branch. Generally it's kinda risky to have this as it's detached from the actual user state (it's separate user) and can lead to issues. Going to remove this and see how we approached logout.